File: EPSILON/TOOLS/Altova/Altova/XMLSpy2006/spl/java/altova/text/edifact/MakeSureHasChild.java
[call GenerateFileHeader("MakeSureHasChild.java")]
package com.altova.text.edifact;

import com.altova.text.*;

class MakeSureHasChild {
	public static ITextNode at(ITextNode parent, String name,
			int possiblepositions\[\]) {
		ITextNode result = NullTextNode.getInstance();
		ITextNodeList children = parent.getChildren();
		for (int i = 0; (i < possiblepositions.length) && (result.isNull()); ++i) {
			int pos = possiblepositions\[i\];
			if (pos > children.size())
				pos = children.size() - 1;
			ITextNode kid = children.getAt(pos);
			if (kid.getName() == name)
				result = kid;
		}
		if (result.isNull()) {
			int pos = possiblepositions\[0\];
			if (pos > children.size())
				pos = children.size();
			result = new TextNode(result, name);
			result.setNativeName(name);
			if (name.equals("GS") || name.equals("GE") || name.equals("ST") || name.equals("SE"))
				result.setNodeClass(ITextNode.Segment);
			else if (name.startsWith("F"))
				result.setNodeClass(ITextNode.DataElement);
			else if (name.startsWith("S"))
				result.setNodeClass(ITextNode.Composite);
			else if ((name.startsWith("U")) && (3 == name.length()))
				result.setNodeClass(ITextNode.Segment);
			else if ((name.startsWith("I")) && (3 == name.length()))
				result.setNodeClass(ITextNode.Segment);
			else
				result.setNodeClass(ITextNode.Group);
			children.insertAt(result, pos);
		}
		return result;
	}

	public static ITextNode at(ITextNode parent, String name, int index) {
		int positions\[\] = new int\[1\];
		positions\[0\] = index;
		return at(parent, name, positions);
	}

	public static ITextNode asFirst(ITextNode parent, String name) {
		return at(parent, name, 0);
	}

	public static ITextNode asLast(ITextNode parent, String name) {
		return at(parent, name, 1000000);
	}
}

File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/cs/AltovaText/EDIFact/MakeSureHasChild.cs
[call GenerateFileHeader("MakeSureHasChild.cs")]
namespace Altova.TextParser.EDIFACT
{
	internal class MakeSureHasChild
	{
		public static ITextNode At(ITextNode parent, string name, int\[\] possiblepositions)
		{
			ITextNode result = NullTextNode.Instance;
			ITextNodeCollection children = parent.Children;
			for (int i = 0; (i < possiblepositions.Length) && (result.IsNull); ++i)
			{
				int pos = possiblepositions\[i\];
				if (pos > children.Count) pos = children.Count - 1;
				ITextNode kid = children\[pos\];
				if (kid.Name == name)
					result = kid;
			}
			if (result.IsNull)
			{
				int pos = possiblepositions\[0\];
				if (pos > children.Count)
					pos = children.Count;
				result = new TextNode(result, name);
				result.NativeName = name;
				if ((name == "GS") || (name == "GE") || (name == "ST")
					|| (name == "SE"))
					result.Class = NodeClass.Segment;
				else if (name.StartsWith("F"))
					result.Class = NodeClass.DataElement;
				else if (name.StartsWith("S"))
					result.Class = NodeClass.Composite;
				else if ((name.StartsWith("U")) && (3 == name.Length))
					result.Class = NodeClass.Segment;
				else if ((name.StartsWith("I")) && (3 == name.Length))
					result.Class = NodeClass.Segment;
				else
					result.Class = NodeClass.Group;
				children.Insert(result, pos);
			}
			return result;
		}
		public static ITextNode At(ITextNode parent, string name, int index)
		{
			int\[\] positions = new int\[1\];
			positions\[0\] = index;
			return At(parent, name, positions);
		}
		public static ITextNode AsFirst(ITextNode parent, string name)
		{
			return At(parent, name, 0);
		}
		public static ITextNode AsLast(ITextNode parent, string name)
		{
			return At(parent, name, 1000000);
		}
	}
}